packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -308,31 +312,62 @@ export const getHeritage = (
 
 export const getHeritageFromExpression = (
   expression: ts.Expression,
-  analyzer: AnalyzerInterface
+  analyzer: AnalyzerInterface,
+  isMixinClass?: boolean
 ): ClassHeritage => {
-  // TODO(kschaaf): Support for extracting mixing applications from the heritage
-  // expression https://github.com/lit/lit/issues/2998
   const mixins: Reference[] = [];
-  const superClass = getSuperClass(expression, analyzer);
+  const superClass = getSuperClassAndMixins(expression, mixins, analyzer);
   return {
-    superClass,
+    superClass: isMixinClass ? undefined : superClass,
     mixins,
   };
 };
 
-export const getSuperClass = (
+export const getSuperClassAndMixins = (
```

Comment:
Nit: Could `getSuperClassAndMixins` return the `mixins` reference array, rather than expecting an empty one and mutating it?

Unless there is a case where a previously populated mixins array should be passed in?

Reply:
I take it back, I like the recursive implementation that this allows.